modules/agentless-s3-bucket/variables.tf

```diff
@@ -0,0 +1,26 @@
+
+variable "tags" {
+  description = "A map of additional tags to add to the IAM role/profile created"
+  type        = map(string)
+  default     = {}
+}
+
+variable "iam_delegate_role_arn" {
+  description = "Reference ARN to the Agentless delegate role"
+  type        = string
+}
+
+variable "iam_rds_assume_role_arn" {
+  description = "Reference ARN to the Agentless RDS assume role"
+  type        = string
+}
+
+variable "primary_kms_key_arn" {
+  description = "Primary KMS key ARN to encrypt the exported data"
+  type        = string
+}
+
+variable "primary_kms_key_region" {
```

Comment:
Instead of getting this as a variable, you can use the [aws_kms_key](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/kms_key) data source. This way you don't even need the primary key ARN. It could be a replica ARN or just the key ID.

Reply:
can't do this, to work, we need to be working under the same provider scope as the one in which we create it.